src/simulator.py

```diff
@@ -10,6 +10,8 @@
 
 from __future__ import print_function
 from __future__ import with_statement
+
+import traceback
```

Comment:
It seems `traceback` package is not used in the script. Maybe we can remove this line?